src/types/presentation.ts

```diff
@@ -0,0 +1,110 @@
+import { StatusCodes } from 'http-status-codes';
+import type { ICommonErrorResponse } from './authentication.js';
+import type {
+	CompactJWT,
+	ContextType,
+	UnsignedPresentation,
+	VerifiablePresentation,
+	W3CVerifiablePresentation,
+} from '@veramo/core';
+import { jwtDecode, type JwtPayload } from 'jwt-decode';
+import type { IIdentityService } from '../services/identity/index.js';
+import type { CustomerEntity } from '../database/entities/customer.entity.js';
+import { CheqdW3CVerifiableCredential } from './credential.js';
+import { CommonReturn } from './shared.js';
+import { JWT_PROOF_TYPE } from './constants.js';
+
+export interface ICheqdPresentation extends UnsignedPresentation {
+	proof: {
+		type: string;
+		jwt: string;
+	};
+}
+
+export interface IJWTPayloadVP extends JwtPayload {
+	vp: ICheqdPresentation;
+}
+
+// ToDo: make unit tests
+export class CheqdW3CVerifiablePresentation extends CommonReturn implements ICheqdPresentation {
```

Comment:
Dont think a class should be under the types folder

Reply:
Where do you think it should be?

Reply:
under /services/presentation/